README.md

```diff
@@ -34,111 +23,164 @@ Automate the process of extracting documentation URLs and adding them as sources
 - **Bulk URL Loading**: Add multiple URLs to NotebookLM automatically
 - **Error Handling**: Comprehensive error messages and recovery options
 - **YouTube & Website Support**: Handles both content types
+- **Smart Version Detection** The script automatically detects version numbers in URLs and handles them intelligently:
+
+    | URL Format | `--versions` Flag | Behavior |
+    |------------|------------------|----------|
+    | `https://docs.example.com/product/3.2` | Not specified | Uses detected version `3.2` |
+    | `https://docs.example.com/product/3.2` | `--versions 2.21,latest` | Ignores detected `3.2`, uses specified versions |
+    | `https://docs.example.com/product` | Not specified | Uses default `latest` |
+    | `https://docs.example.com/product` | `--versions 2.21,latest` | Uses specified versions |
 
-## Quick Start
+- **Supported Version Patterns**: Includes `/latest`, `/3.2`, `/v3.2`, `/2.21.1`
+- **Consistent File Handling**: The script uses **predictable file handling** for easier workflows:
+  - **Extraction**: Always saves to `urls.txt` (unless `--toc-output` specified)
+  - **Notebook Mode**: Always reads from `urls.txt` (unless `--links-file` specified)
+  - **Resource Combination**: Always includes `CQA_res.txt` static resources (unless `--skip-cqa` is used)
+  - **No Guessing**: Clear, consistent behavior every time
+
+## Installation
 
 ### Prerequisites
 - Python 3.7+
 - A NotebookLM account
 
-### Installation
+### Procedure
+
+1. OPTIONAL: If the project root is different from the script directory `add_links_notebook`, navigate to the location where the `.venv` directory is located:
+  ```bash
+  # Navigate to project root (where .venv is located) 
+  cd ../add_links_notebook
+  ```
+
+2. Create a virtual environment if it is not available. 
+  ```bash
+  # If using this for the first time, create a new virtual environment
+  python3 -m venv .venv
+  ```
+3. Activate the virtual environment:
+  ```bash
+  # Activate the existing virtual environment
+  source .venv/bin/activate
+  ```
+4. Install dependencies.
+  ```bash
+  python3 -m pip install -r requirements.txt
+  ```
+
+5. Install browser binaries for Playwright.
+  ```bash
+  python3 -m playwright install
+  ```
+
+6. Install the Playwright browser.
+  ```bash
+  # Check Playwright is installed
+  playwright --version
+  ```
+7. Install Chromium browser to automate authentication.
+  ```bash
+  playwright install chromium
+  ```
+
+8. OPTIONAL: If your project root is different from the script directory, navigate back to the script directory:
+  ```bash
+  # Navigate back to script directory
+  cd ../add_scrapped_links_notebooklm
+  ```
```

Comment:
I think this is related to step 1. Users would need to run this step only if they are in a directory that does not contain the script.

Reply:
Maybe, we can make it a mandatory step in this format:
Ensure to check whether the current directory is the same directory in which the script is placed so that you can run it.